packages/flutter/lib/src/widgets/display_feature_sub_screen.dart

```diff
@@ -0,0 +1,239 @@
+  final EdgeInsets? padding;
+
+  @override
+  Widget build(BuildContext context) {
+    final Rect availableSpace = _availableSpace(context);
+    final Iterable<Rect> safeAreas = _safeAreasInBounds(context, availableSpace);
+    final Rect safeArea = anchorPoint == null
+        ? _firstSafeArea(safeAreas, context)
+        : _closestToAnchorPoint(safeAreas, _finiteOffset(anchorPoint!));
+
+    return Align(
+      alignment: Alignment.topLeft,
+      child: Padding(
+        padding: EdgeInsets.only(left: safeArea.left, top: safeArea.top),
+        child: SizedBox(
+          width: safeArea.width,
+          height: safeArea.height,
+          child: child,
+        ),
+      ),
+    );
+  }
+
+  static Rect _firstSafeArea(Iterable<Rect> safeAreas, BuildContext context) {
+    final TextDirection textDirection = Directionality.of(context);
+    switch (textDirection) {
+      case TextDirection.rtl:
+        return _closestToAnchorPoint(safeAreas, const Offset(double.maxFinite, 0));
+      case TextDirection.ltr:
+        return _closestToAnchorPoint(safeAreas, Offset.zero);
+    }
+  }
+
+  static Rect _closestToAnchorPoint(Iterable<Rect> safeAreas, Offset anchorPoint) {
+    return safeAreas.fold(safeAreas.first, (Rect previousValue, Rect element) {
+      final double previousDistance = _distanceFromPointToRect(anchorPoint, previousValue);
+      final double elementDistance = _distanceFromPointToRect(anchorPoint, element);
+      if (previousDistance < elementDistance)
+        return previousValue;
+      else
+        return element;
+    });
+  }
+
+  static double _distanceFromPointToRect(Offset point, Rect rect){
+    // Cases for point position relative to rect:
+    // 1  2  3
+    // 4 [R] 5
+    // 6  7  8
+    if (point.dx < rect.left) {
+      if (point.dy < rect.top) {
+        // Case 1
+        return (point - rect.topLeft).distance;
+      } else if (point.dy > rect.bottom){
+        // Case 6
+        return (point - rect.bottomLeft).distance;
+      } else {
+        // Case 4
+        return rect.left - point.dx;
+      }
+    } else if (point.dx > rect.right) {
+      if (point.dy < rect.top) {
+        // Case 3
+        return (point - rect.topRight).distance;
+      } else if (point.dy > rect.bottom){
+        // Case 8
+        return (point - rect.bottomRight).distance;
+      } else {
+        // Case 5
+        return point.dx - rect.right;
+      }
+    } else {
+      if (point.dy < rect.top) {
+        // Case 2
+        return rect.top - point.dy;
+      } else if (point.dy > rect.bottom){
+        // Case 7
+        return point.dy - rect.bottom;
+      } else {
+        // Case R
+        return 0;
+      }
+    }
+  }
+
+  Rect _availableSpace(BuildContext context){
+    if (padding != null) {
+      return padding!.deflateRect(Offset.zero & MediaQuery.of(context).size);
+    } else {
+      final RenderObject? renderObject = Overlay.of(context)?.context.findRenderObject();
```

Comment:
I agree. Is there a good example I can look at for these scenarios?

Reply:
@goderbauer What I have trouble with in particular is trying to call `getTransformTo(null)` during `performLayout()`. This render object needs to know it's own position on the screen in order to position its child. Is that possible?

Reply:
I updated this following your suggestion. Please have another look.

I have a feeling it's still not ok, since the solution I adopted has situations where it skips a frame, but the code is better suited for you to review again. I will write a separate comment with everything I've tried, but the short version is that the `Constraints go down. Sizes go up. Parent sets position` flutter layout algoritm makes it impossible to know the global position during layout. The global position is needed for calculating how display features intersect with the current render object.

As a solution, I am caching the position during paint and if it changes, I trigger another layout but I am unhappy with the solution and any suggestion is greatly appreciated.